### PR TITLE
Fix prepare_next_version lane and tagging latest tag

### PR DESCRIPTION
**Pull request by vegaro**
There were issues when creating the pull request with the bump to next version so I explicitely added some "optional" parameters to the lane and now it's working fine.

The tagging of the latest release was also broken, it was missing an early exit and had some issues switching back to the current branch.

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -41,17 +41,18 @@ platform :android do
 
   desc "Tag release version with latest if necessary"
   lane :tag_release_with_latest_if_needed do |options|
-    release_version = options[:release_version]
+    release_version = options[:version]
+    puts "Release version is #{release_version}"
 
-    current_branch = git_branch
     sh("git", "checkout", "tags/latest")
     latest_version = android_get_version_name(gradle_file: gradle_file_path)
```
Comment (beylmk) [inline fastlane/Fastfile:48]: can we add a comment about whether we've assuming we've already updated version in gradle or not when we call this? that always trips me up

File: fastlane/Fastfile [hunk 2]
```diff
@@ -140,7 +141,11 @@ platform :android do
     bump(version: next_version_snapshot)
 
     sh("git", "commit", "-am", "Preparing for next version")
-    push_to_git_remote
+    push_to_git_remote(
```
Comment (beylmk) [inline fastlane/Fastfile:144]: 👍

File: fastlane/Fastfile [hunk 3]
```diff
@@ -89,7 +90,7 @@ platform :android do
       upload_assets: []
     )
 
-    tag_release_with_latest_if_needed(release_version: release_version)
+    tag_release_with_latest_if_needed(options)
```
Comment (beylmk) [inline fastlane/Fastfile:93]: just curious, why this update? we're not passing anything besides release_version right?

**Review by beylmk — APPROVED** [review]
thanks! nothing blocking

File: fastlane/Fastfile (hunk 3 above)
Comment (vegaro) [reply fastlane/Fastfile:93]: Yeah, this change didn't do anything. Should I revert it?

Comment (beylmk) [reply fastlane/Fastfile:94]: i think i prefer the more specific naming. but up to you, we do use options elsewhere in the file 🤷

**Pull request merged**